e2e_playwright/st_toast_test.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+def test_default_toast_rendering(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that toasts are correctly rendered via screenshot."""
+    themed_app.keyboard.type("r")
+    toasts = themed_app.get_by_test_id("stToast")
+    expect(toasts).to_have_count(2)
+
+    assert_snapshot(toasts.nth(0), name="toast-regular")
+
+
+def test_collapsed_toast_rendering(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test collapsed long toasts are correctly rendered via screenshot."""
+    themed_app.keyboard.type("r")
```

Comment:
nit: Maybe you can try:

```python
page.get_by_test_id("stApp").click(position={"x": 0, "y": 0})
themed_app.keyboard.press("r")
```

Reply:
another reason why they might be flaky could also be that the toast is not yet fully shown on the UI when `assert_snapshot` is triggered. We had similar flakiness with dialogs. The way how to fix this is by adding an expect's to check that it is actually visible and showing content:

```python
expect(toasts.nth(1)).to_be_visible()
expect(toasts.nth(1)).to_have_text("Random toast message that is a really really really really really really really long message, going way past the 3 line limit" )
```